internal/tool/ssh/impl/babysitter.go

```diff
@@ -108,12 +104,18 @@ func RunBabysitter(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	b.envelope = e
-	go b.collectMetrics()
-	return e.Run(b.ctx)
+	c := metricsCollector{logger: b.logger, envelope: e, info: info}
+	go c.run(ctx)
+	return e.Run(ctx)
 }
 
-func (b *babysitter) collectMetrics() {
+type metricsCollector struct {
+	logger   logtype.Logger
+	envelope *envelope.Envelope
+	info     *BabysitterInfo
+}
+
+func (b *metricsCollector) run(ctx context.Context) {
```

Comment:
Oops, forgot to mention we should probably rename this `b` to an `m`.

Reply:
Sounds good, I'll open a PR with this change.